pkg/clients/database/postgresql.go

```diff
@@ -172,9 +176,10 @@ func (c *PostgreSQLServerClient) UpdateServer(ctx context.Context, cr *azuredbv1
 	// we don't support that.
 	s := cr.Spec.ForProvider
 	properties := &postgresql.ServerUpdateParametersProperties{
-		Version:           postgresql.ServerVersion(s.Version),
-		MinimalTLSVersion: postgresql.MinimalTLSVersionEnum(s.MinimalTLSVersion),
-		SslEnforcement:    postgresql.SslEnforcementEnum(s.SSLEnforcement),
+		Version:             postgresql.ServerVersion(s.Version),
+		MinimalTLSVersion:   postgresql.MinimalTLSVersionEnum(s.MinimalTLSVersion),
+		SslEnforcement:      postgresql.SslEnforcementEnum(s.SSLEnforcement),
+		PublicNetworkAccess: postgresql.PublicNetworkAccessEnum(azure.ToString(s.PublicNetworkAccess)),
```

Comment:
Did you do some manual tests to ensure that Azure actually supports changing the public access mode of an existing PostgreSQL server?

Reply:
Yes, I did manual testing for this case and Azure allows and updates this field in an existing PostgreSQL server